service/controller/azureconfig/handler/deployment/resource.go

```diff
@@ -431,7 +431,7 @@ func (r *Resource) setResourceStatus(ctx context.Context, customObject providerv
 	}
 
 	{
-		err := r.ctrlClient.Update(ctx, &customObject)
+		err := r.ctrlClient.Status().Update(ctx, &customObject)
```

Comment:
Fix bug from base branch (this PR https://github.com/giantswarm/azure-operator/pull/1643)